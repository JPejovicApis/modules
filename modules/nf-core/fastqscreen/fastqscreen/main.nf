process FASTQSCREEN_FASTQSCREEN {
    tag "$meta.id"
    label 'process_medium'

    conda "bioconda::fastq-screen=0.15.3"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/fastq-screen%3A0.15.3--pl5321hdfd78af_0':
        'quay.io/biocontainers/fastq-screen:0.15.3--pl5321hdfd78af_0'}"

    input:
    tuple val(meta), path(reads)  // .fastq files
    path database

    output:
    tuple val(meta), path ("*_fq_screen"), emit: fastq_screen
    path "versions.yml"                  , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ""

    """
    fastq_screen --threads ${task.cpus} \\
        --aligner bowtie2 \\
        --conf ${database}/fastq_screen.conf \\
        $reads \\
        $args \\
        --outdir ${prefix}_fq_screen

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastqscreen: \$(echo \$(fastq_screen --version 2>&1) | sed 's/^.*FastQ Screen v//; s/ .*\$//')
    END_VERSIONS
    """
}
