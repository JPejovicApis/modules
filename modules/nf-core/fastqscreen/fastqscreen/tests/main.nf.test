nextflow_process {

    name "Test Process FASTQSCREEN_FASTQSCREEN"
    script "../main.nf"
    process "FASTQSCREEN_FASTQSCREEN"

    tag "modules"
    tag "modules_nfcore"
    tag "bowtie2/build"
    tag "fastqscreen"
    tag "fastqscreen/buildfromindex"
    tag "fastqscreen/fastqscreen"

    test("sarscov2 - human") {

        setup {

            run("BOWTIE2_BUILD") {
                script "../../../bowtie2/build/main.nf"
                process {
                    """
                    input[0] =  Channel.from([
                                        [[id: "sarscov2"], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)],
                                        [[id: "human"]   , file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)]
                                ])
                    """
                }
            }

            run("FASTQSCREEN_BUILDFROMINDEX") {
                script "../../../fastqscreen/buildfromindex/main.nf"
                process {
                    """
                    input[0] = BOWTIE2_BUILD.out.index.map{meta, index -> meta.id}.collect()
                    input[1] = BOWTIE2_BUILD.out.index.map{meta, index -> index}.collect()
                    """
                }
            }

        }

        when {
            process {
                """
                input[0] =  [[ id:'test', single_end:true ],
                             [file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
                            ]
                input[1] = FASTQSCREEN_BUILDFROMINDEX.out.database
                """
            }
        }

        then {
            assertAll(
                { assert process.success }
            )
        }

    }

    test("sarscov2 - human - stub") {

        options "-stub"

        setup {

            run("BOWTIE2_BUILD") {
                script "../../../bowtie2/build/main.nf"
                process {
                    """
                    input[0] =  Channel.from([
                                        [[id: "sarscov2"], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)],
                                        [[id: "human"]   , file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)]
                                ])
                    """
                }
            }

            run("FASTQSCREEN_BUILDFROMINDEX") {
                script "../../../fastqscreen/buildfromindex/main.nf"
                process {
                    """
                    input[0] = BOWTIE2_BUILD.out.index.map{meta, index -> meta.id}.collect()
                    input[1] = BOWTIE2_BUILD.out.index.map{meta, index -> index}.collect()
                    """
                }
            }

        }

        when {
            process {
                """
                input[0] =  [[ id:'test', single_end:true ],
                             [file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
                            ]
                input[1] = FASTQSCREEN_BUILDFROMINDEX.out.database
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
