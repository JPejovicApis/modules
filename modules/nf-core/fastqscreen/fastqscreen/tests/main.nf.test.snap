{
    "sarscov2 - human": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1_screen.txt:md5,b1fde8e1b4b792f6bc001f027368e067"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1_screen.png:md5,ba6ba10dcf4667379289a7f4d2d007c5"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T13:50:32.428126236"
    },
    "sarscov2 - human - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1_screen.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1_screen.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1_screen.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,8ac0239b5103352958d9a9e562b23103"
                ],
                "html": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1_screen.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "png": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1_screen.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1_screen.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8ac0239b5103352958d9a9e562b23103"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T13:50:45.968423801"
    }
}